### ISSUE TITLE
Social icons block: remove the CSS order property

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#58938]
### Description

For accessibility, visual order and DOM order must always match when they affect 'meanning and operation'. Reference:

WCAG 2.2
1.3.2 Meaningful Sequence (Level A) https://www.w3.org/TR/WCAG22/#meaningful-sequence
2.4.3 Focus Order (Level A) https://www.w3.org/TR/WCAG22/#focus-order

The Social Icons block does use the CSS `order` property, which seems to be unnecessary now. 
Introduced in https://github.com/WordPress/gutenberg/commit/d0f4a63d739b06e5f5051ea2d9880b63e3e971fc

Relevant code:

https://github.com/WordPress/gutenberg/blob/f1cd56679bc926ad1d9ec86060616c3579efe95e/packages/block-library/src/social-links/editor.scss#L77

At that time, in WordPress 5.8, visually, the text 'Click plus to add' was on the right and the plus was on the lieft. The `order` property actually visually swapped the two elements:

![5 8](https://github.com/WordPress/gutenberg/assets/1682452/7eca7178-b410-48bb-8c1c-656409551d33)

Something has changed in the meantime, as the plus is now absolutely positioned and it shows at the right end of the block. As such, the `order` property doesn't do anything visually.

![Screenshot 2024-02-12 at 14 26 27](https://github.com/WordPress/gutenberg/assets/1682452/1af522f4-d87a-491d-8a6b-2b9c0a737646)

I'm not sure this is the intended design. Cc @WordPress/gutenberg-design 
Anyways, it appears there's some lack of understanding about the a11y rule to not change the order. The comment in the code:

`Move after the appender button. Because this element is non-interactive,it does not affect tab order.`

seems to hint that changing the order is allowed when the elements are non-interactive. That's not correct. The order must be preserved also for non-interactive elements because the _reading_ order must match the DOM order.
To clarify: it's not just about the tabbing order. It's about the visual/reading order and the DOM order.

In any case:
- If the design must be restored to what it was in WordPress 5.8, an alternative solution must be used. We can't use 'order'.
- If the new design is the intended one, with the plus at the right, the `order` property is unnecessary and can be safely removed.

Cc @mkaz  

### Step-by-step reproduction instructions

- Add a Social icons block.
- Observe the block shows a placeholder text 'Click plus to add'.
- Inspect that text in your browser dev tools.
- Select the element with CSS class `wp-block-social-links__social-prompt`.
- Observe this element has a CSS property `order: 2;`.
- Unset the property and observe nothing changes.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
